test/OpenTracing.Tests/Util/AsyncLocalScopeManagerTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using NSubstitute;
+using OpenTracing.Util;
+using Xunit;
+
+namespace OpenTracing.Tests.Util
+{
+    public class AsyncLocalScopeManagerTests
+    {
+        private AsyncLocalScopeManager _source;
+
+
+        public AsyncLocalScopeManagerTests()
+        {
+            _source = new AsyncLocalScopeManager();
+        }
+
+        [Fact]
+        public void MissingActiveScope()
+        {
+            IScope missingScope = _source.Active;
+            Assert.Null(missingScope);
+        }
+
+        [Fact]
+        public void DefaultActivate()
+        {
+            ISpan span = Substitute.For<ISpan>();
+
+            using (IScope scope = _source.Activate(span, false))
+            {
+                Assert.NotNull(scope);
+                IScope otherScope = _source.Active;
+                Assert.Same(otherScope, scope);
+            }
+
+            // Make sure the span is not finished.
+            span.DidNotReceive().Finish();
+
+            // And now it's gone:
+            IScope missingScope = _source.Active;
+            Assert.Null(missingScope);
+        }
+
+        [Fact]
+        public void FinishSpanOnDispose()
+        {
+            ISpan span = Substitute.For<ISpan>();
+
+            using (IScope scope = _source.Activate(span, true))
+            {
+                Assert.NotNull(scope);
+                Assert.NotNull(_source.Active);
+            }
+
+            // Make sure the ISpan got finish()ed.
+            span.Received(1).Finish();
+
+            // Verify it's gone.
+            Assert.Null(_source.Active);
+        }
+
+        [Fact]
+        public void DontFinishSpanNoDispose()
+        {
+            ISpan span = Substitute.For<ISpan>();
+
+            using (IScope scope = _source.Activate(span, false))
```

Comment:
[nit] I generally prefer named args for booleans (or nulls) because it makes the call site make a lot more sense.
E.g. in this place you'd be able to see we expect no `Finish()` because we said `finishOnDispose: false`
Without it you require more contextual brain-knowledge to be able to understand the test

Reply:
fixed